python/ccxt/base/exchange.py

```diff
@@ -853,7 +869,7 @@ def safe_value_n(dictionary, key_list, default_value=None):
 
     @staticmethod
     def get_object_value_from_key_list(dictionary, key_list):
-        filtered_list = list(filter(lambda el: el in dictionary, key_list))
+        filtered_list = list(filter(lambda el: el in dictionary and dictionary[el] != '' and dictionary[el] is not None, key_list))
```

Comment:
@samgermain why is this needed? can you give us a real-case scenario?

Reply:
These are the examples, I'll need to go to a PR I did about 6 months ago to find the real use case, I fixed the problem for `safe_string_2` but not `safe_string_n`

```python
safe_string_lower_n ({'test': '', 'test2': '10'}, ['test', 'test2'])    # should return 10, actually returns ''
safe_string_upper_n ({'test': '', 'test2': '10'}, ['test', 'test2'])    # should return 10, actually returns ''
safe_string_lower_n ({'test': ''}, ['test', 'test2'])                   # should return None, actually returns ''
safe_string_upper_n ({'test': ''}, ['test', 'test2'])                   # should return None, actually returns ''
safe_string_lower_n ({'test2': ''}, ['test', 'test2'])                  # should return None, actually returns ''
safe_string_upper_n ({'test2': ''}, ['test', 'test2'])                  # should return None, actually returns ''
```

Reply:
[Here's the commit where it was fixed for `safeString2`](https://github.com/ccxt/ccxt/pull/13438)

It looks like [OKX stop orders were having issues with dictionary/object values being empty strings](https://github.com/ccxt/ccxt/pull/13440)

Reply:
@samgermain Yeah okx might have a weird api but I'm not sure we should do that by default.

For instance:
```Javascript
safe_string_lower_n ({'test': ''}, ['test', 'test2'], "defaultValue") // will return "defaultValue"
```   
⬆️ why should we return the defaultValue here and hide the key existence?

Reply:
> @samgermain Yeah okx might have a weird api but I'm not sure we should do that by default.
> 
> For instance:
> 
> ```js
> safe_string_lower_n ({'test': ''}, ['test', 'test2'], "defaultValue") // will return "defaultValue"
> ```
> 
> ⬆️ why should we return the defaultValue here and hide the key existence?

Because there isn't a value, if we need a number for string math or something, we would need to perform the math on either the value that the exchange gives us, or on the default value, but we don't want to perform it on an empty string, which is the absence of a value

Reply:
I moved this change to a separate PR https://github.com/ccxt/ccxt/pull/15854

Reply:
@samgermain thanks we can keep this discussion there then.